src/lib/site-server-process.ts

```diff
@@ -142,12 +142,33 @@ export default class SiteServerProcess {
 		}
 
 		return new Promise< void >( ( resolve, reject ) => {
+			const KILL_TIMEOUT = 5000;
+			let isResolved = false;
+
+			const forceKillTimeout = setTimeout( () => {
+				if ( isResolved ) return;
+
+				// Check if process and pid are still valid before attempting force kill
+				if ( process && process.pid ) {
+					require( 'process' ).kill( process.pid, 'SIGKILL' );
+				} else {
+					if ( ! isResolved ) {
+						isResolved = true;
+						resolve();
+					}
+				}
```

Comment:
```suggestion
				if ( process.pid ) {
					kill( process.pid, 'SIGKILL' );
				} else {
					isResolved = true;
					resolve();
				}
```
Nit, but we already know that `process` is defined at this point, and we also know that `isResolved` is false because this callback returns early if that's the case.

I also don't see a good reason to use the `require` pattern here – we might as well do `import { kill } from 'process';`